### DISCUSSION TITLE
How to generate Description Attribute for enum by IIncrementalGenerator

### DISCUSSION THREAD

**Discussion by LGinC** [dotnet/roslyn#69984] [General]
MyState.cs
```CSharp
public enum MyState
{
    /// <summary>
    ///  StartOfWorkflow
    /// </summary>
    Start,

    /// <summary>
    ///  EndOfWorkflow
    /// </summary>
    End,
}
```

and  I want to modify MyState.cs to 
```CSharp
public enum MyState
{
    /// <summary>
    ///  StartOfWorkflow
    /// </summary>
    [Description("StartOfWorkflow")]
    Start,

    /// <summary>
    ///  EndOfWorkflow
    /// </summary>
    [Description("EndOfWorkflow")]
    End,
}
```

just add `[Description("xxx")]` from comment

how to do this by IIncrementalGenerator？

**Comment by CyrusNajmabadi** [comment]
You cannot do this. Generators can only add new files. They cannot modify existing files.

**Comment by KennethHoff** [comment]
What you could do is create the enum in it's entirely. Instead of trying to add something to the enum based on a comment, create the enum from the comment - that'd allow you to add the Attribute
